boards/tenstorrent/tt_blackhole/tt_blackhole_fixed_partitions.dtsi

```diff
@@ -14,101 +20,171 @@ partitions {
 	};
 
-	flshinfo: partition@59000 {
-		label = "flshinfo";
-		reg = <0x59000 DT_SIZE_K(4)>;
-		binary-path =
-			"$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/flash_info.bin";
-		read-only;
+	/*
+	 * Image trailer for the DMC firmware update. Indicates to MCUBoot that
+	 * the image is valid and should be booted in test mode.
+	 */
+	dmfw_trailer: partition@c9000 {
+		label = "dmfw_trailer";
+		reg = <0xc9000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/mcuboot_magic_test.bin";
 	};
 
-	boardcfg: partition@5a000 {
-		label = "boardcfg";
-		reg = <0x5a000 DT_SIZE_K(4)>;
-		binary-path = "$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/read_only.bin";
-		read-only;
+	/*
+	 * This partition is used for the primary firmware image.
+	 * it is loaded by MCUBoot. It is tagged as
+	 * mainimg in tt-boot-fs, but isn't used by the bootrom.
+	 */
+	slot0_partition: partition@ca000 {
+		label = "mainimg";
+		reg = <0xca000 DT_SIZE_K(512)>;
+		binary-path = "$BUILD_DIR/smc/zephyr/zephyr.signed.bin";
 	};
 
 	/*
-	 * FIXME: The address of this node triggers a warning that we should try to fix:
-	 * Warning (unique_unit_address_if_enabled):
-	 *  /soc/spi@80070000/eeprom@0/partitions/partition@10000000
-	 * duplicate unit-address (also used in node
-	 *   /soc/spi@80070000/eeprom@0/partitions/failover_partition@10000000)
+	 * This partition defines a binary blob that indicates the
+	 * main image should be booted in test mode by mcuboot.
 	 */
-	failover: failover_partition@10000000 {
+	main_test: partition@149000 {
+		label = "maintail";
+		reg = <0x149000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/mcuboot_magic_test.bin";
+	};
+
+	/*
+	 * All partitions below here should eventually be mapped to binaries.
+	 * These binaries will not change, and should provide a safe bootloader
+	 * path in the event of an invalid main image.
+	 */
+
+	/*
+	 * Install MCUBoot as the cmfw tag as well as the failover. Otherwise
+	 * the bootrom may fail to load, as it expects a cmfw tag to be
+	 * present.
+	 */
+	boot_partition: partition@14a000 {
+		label = "cmfw";
+		reg = <0x14a000 DT_SIZE_K(128)>;
+		binary-path = "$BUILD_DIR/mcuboot/zephyr/zephyr.bin";
+	};
+
+	/*
+	 * This partition contains the recovery image. It will be
+	 * loaded by MCUBoot if the primary image fails to validate.
+	 * This binary is not used by the bootrom.
+	 */
+	slot1_partition: partition@16a000 {
+		label = "safeimg";
+		reg = <0x16a000 DT_SIZE_K(512)>;
+		binary-path = "$BUILD_DIR/recovery/zephyr/zephyr.signed.bin";
+	};
+
+	/*
+	 * This partition defines a binary blob that indicates the
+	 * recovery image is valid and bootable to mcuboot.
+	 */
+	recovery_valid: partition@1e9000 {
+		label = "safetail";
+		reg = <0x169000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/mcuboot_magic_confirmed.bin";
+	};
+
+	/*
+	 * MCUBoot partition- will be loaded by bootrom.
+	 * tagged as fail_over_image in tt-boot-fs,
+	 * but we actually use it as the primary image.
+	 */
+	boot_fallback: partition@1ea000 {
```

Comment:
Since we tag mcuboot as the CMFW image and failover image, we actually should be fine to continue using the tt-boot-fs in our firmware application. This is because even if we write an invalid tt-boot-fs, the failover image header is in a separate page and thus won't be touched- so the bootrom *should* fall back to the other copy of mcuboot.

We don't have to do this though- it may be safer to instead use a different copy of the tt-boot-fs within our application